felix/bpf/conntrack/scanner.go

```diff
@@ -156,14 +167,29 @@ func (s *Scanner) Scan() {
 				"entry": ctVal,
 			}).Debug("Examining conntrack entry")
 		}
+		val = cleanupv1.Value{}
+		if s.ipVersion == 6 {
+			val = cleanupv1.ValueV6{}
+		}
 
 		for _, scanner := range s.scanners {
-			if verdict := scanner.Check(ctKey, ctVal, s.get); verdict == ScanVerdictDelete {
+			if verdict, ts := scanner.Check(ctKey, ctVal, s.get); verdict == ScanVerdictDelete {
 				if debug {
 					log.Debug("Deleting conntrack entry.")
 				}
-				cleaned++
-				return maps.IterDelete
+				revKey := []byte{}
+				if ctVal.Type() == TypeNATForward {
+					revKey = ctVal.ReverseNATKey().AsBytes()
+				}
+
+				val = cleanupv1.NewValue(revKey, uint64(ctVal.LastSeen()), uint64(ts))
+				if s.ipVersion == 6 {
+					val = cleanupv1.NewValueV6(revKey, uint64(ctVal.LastSeen()), uint64(ts))
+				}
+				err := s.ctCleanupMap.Update(ctKey.AsBytes(), val.AsBytes())
+				if err != nil {
+					log.Debugf("error updating cleanup map. Entry may not be cleaned up %v", err)
```

Comment:
An obvious potential failure is if the cleaner map gets full.

Rather than waiting until the end of the iteration, what if we triggered the BPF cleaner whenever we've got at least N (say 1000) entries ready to go (and then once at the end)?